File: 프로그래머스/lv2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        //f(1) = 1;
        //f(2) = 2;
        //f(3) = 3;
        //f(4) = 5;
        //f(n) = f(n-1) + f(n-2);
    long[] dp = new long[n + 1];
    dp[0] = 1;
    dp[1] = 1;

        for (int i = 2; i <= n; i++)
	dp[i] = (dp[i - 1] + dp[i - 2]) % 1234567;

return dp[n];
}
}

File: 프로그래머스/lv2/12914. 멀리 뛰기/멀리 뛰기.py

def solution(n):
    f = [0] * (n + 1)  # Initialize the array to store Fibonacci numbers 
    f[0] = 1
    f[1] = 1
    if n < 2 :
        return f[n]
    for i in range(2, n+1):  # Changed loop condition and range
        f[i] = f[i - 1] + f[i - 2]
        f[i] = f[i] % 1234567

    return f[n]